### PR TITLE
Add Thanos Ruler service monitor

### PR DESCRIPTION
**Pull request by kakkoyun**
This PR,
* adds Thanos Ruler service monitor
* refactors others to use templating

xref: #123

### CODE REVIEW

File: environments/sre/servicemonitors/observatorium-jaeger.servicemonitor.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ spec:
   - port: admin-http
   namespaceSelector:
     matchNames:
-    - telemeter-production
+    - '{{namespace}}'
```
Comment (squat) [inline environments/sre/servicemonitors/observatorium-jaeger.servicemonitor.yaml]: Hmm I’m not convinced about putting more layers of te playing into this repo. This is no longer usable by the community and also means we cannot use this manifests in e2e tests

Comment (kakkoyun) [reply environments/sre/servicemonitors/observatorium-jaeger.servicemonitor.yaml]: Good point. I'm going to create a dedicated directory for this specific use case and leave others as is.

Comment (kakkoyun) [reply environments/sre/servicemonitors/observatorium-jaeger.servicemonitor.yaml]: I've ended up making them parametric so anyone can easily override `environment` and `namespace` values downstream.

**Comment by kakkoyun** [comment]
As a follow up to this, I'll create another PR to refactor out service monitors as we did in #144, then we won't need additional templating.

**Comment by kakkoyun** [comment]
ping @metalmatze @squat Let's merge this and create another one to move service monitors as #144

File: environments/sre/servicemonitors/observatorium-jaeger.servicemonitor.yaml (hunk 1 above)
Comment (metalmatze) [reply environments/sre/servicemonitors/observatorium-jaeger.servicemonitor.yaml]: I agree with @squat. The environment should be obsolete entirely, as the Prometheus scrape only their environment now.

**Comment by squat** [comment]
rock on

**Pull request merged**